src/planner/binder/query_node/plan_select_node.cpp

```diff
@@ -96,6 +95,14 @@ unique_ptr<LogicalOperator> Binder::CreatePlan(BoundSelectNode &statement) {
 			PlanSubqueries(&expr, &root);
 		}
 		D_ASSERT(!unnest->expressions.empty());
+		if (root->type == LogicalOperatorType::LOGICAL_WINDOW) {
+			for (auto &expression : root->expressions) {
+				if (expression->type != ExpressionType::WINDOW_ROW_NUMBER) {
+					throw BinderException(
+					    "Cannot have window expression over an unlist unless the window funciton is row_number()");
```

Comment:
I assume the reason `row_number` is exempt is because it does not have any arguments, which causes no issues to arise? Would it not be possible to instead look at the actual arguments to check if an issue would arise, rather than making a specialization on row number here?

Reply:
Yea, row number is exempt for that reason.

I can check the arguments to make sure nothing is bound to the unnest alias. Is there a clean way to get the alias string of the unnest? I have access to the unnest object, but for some reason the alias for the unnest expression is not initialized.

There are other problems with this though. In the last test I have 
`SELECT unlist(str_split('A',' ')) AS "r", CASE WHEN ("r" = 'A') THEN (-1) ELSE 0 END AS "x", sum("x") OVER () AS "s";`

the window is over an alias to an alias of an unlist, so I would also have to check for that case.

Also, when I call `EnumerateChildren` to check the argument types, the function doesn't then recursively enumerate the children. Is this intended behavior? If I have a the test case  `select unnest([5,4,3]) as r, sum(r + 42) OVER ();` it will pass since the children of the `bound_function` expression are not checked. 

I can probably find ways to solve each of the above issues, but I think just prohibiting it in the current way is cleaner. Users can still put unnest statements in the `from` clause

Reply:
It might be easier to look at the number of elements in the `BoundSelectNode::unnests` list rather than looking at the children. If the number increases then `UNNEST` clauses have been bound.

Reply:
So I don't think the number of children in the unnest increases when binding the window because the unnest clause comes first in the select list and is therefore bound first. However, during the window binding I can check what ColumnRef alias the window is bound to. 

I added some functionality to check for that and I think the current solution is a bit more elegant.